tests/10apidoc/01register.pl

```diff
@@ -126,7 +128,75 @@ sub matrix_register_user
    });
 }
 
-push @EXPORT, qw( local_user_fixture local_user_fixtures );
+push @EXPORT, qw( matrix_register_user_via_secret );
+
+sub matrix_register_user_via_secret
+{
+   my ( $http, $uid, %opts ) = @_;
+
+   my $password = $opts{password} // "an0th3r s3kr1t";
+   my $admin = $opts{admin} // 0;
+
+   defined $uid or
+      croak "Require UID for matrix_register_user";
+
+   my $hmac = Digest::HMAC_SHA1->new("reg_secret");
+   $hmac->add($uid);
+   $hmac->add($password);
+   $hmac->add($admin ? "admin" : "notadmin");
```

Comment:
Also, more of a spec thought - isn't it usual to include some sort of separator character between these fields, so as to avoid certain collision attacks?